src/openqaoa-core/openqaoa/algorithms/qaoa/qaoa_workflow.py

```diff
@@ -450,6 +450,21 @@ def evaluate_circuit(
         # if the backend is the analytical simulator, we just return the expectation value of the cost Hamiltonian
         if isinstance(self.backend, QAOABackendAnalyticalSimulator):
             output_dict.update({"cost": self.backend.expectation(params_obj)[0]})
+        # if the workflow implements SPAM Twirling, 
+        # we just return the expectation value of the cost Hamiltonian and measurement outcomes
+        elif isinstance(self.backend, SPAMTwirlingWrapper):
+            cost = self.backend.expectation(params_obj)
+            measurement_results = (
+                self.backend.measurement_outcomes
```

Comment:
also,` measurement_results = ...` is repeated, so we can put it outside the if/else block

Reply:
@KilianPoirier, any updates on this?

Reply:
`measurement_result` has to be calculated after calling `expectation()` or `expectation_with_uncertainty()` which means that combining the two would result in either an additional `if`, `else` statement or an additional dictionary update. Here we can clearly distinguish that for different backends we calculate different quantities and the result dictionary is only updated once.